### PR TITLE
Create Restful API for trips

### PR DESCRIPTION
**Pull request by awaisdar001**
This PR adds an api endpoint to the trip modal for curd operations. I have used simplist api using `viewset` and currently it is open for everyone, no restrictions applied.
I also refactored urls to their own apps.
renamed `reactjs` folder to `frontend` as we will be using react and redux here.
Updated requirements.text and installed apps in settings to include rest framework.

### ReadOnly API
`GET` : `localhost:8004/api/trips/` => Get list of all trips
`GET` : `localhost:8004/api/trips/1` => Get detail of one trip

Prerequisites  : 

- [x]  Run `make test` passed?

### CODE REVIEW

**Comment by awaisdar001** [comment]
PR updated, Can @DawoudSheraz / @asadazam93 take a look?

File: djreact/djangoapps/api/serializers.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+class TripListSerializer(serializers.ModelSerializer):
+    """
+    Trip List Modal Serializer
+
+    This Serializer is used for the trips listing as it requires
+    minimal information
+    """
+    activities = ActivitySerializer(many=True)
+    facilities = FacilitySerializer(many=True)
```
Comment (DawoudSheraz) [inline djreact/djangoapps/api/serializers.py]: I think including activities & facilities, which are M2M fields, on the listSerializer is not a good idea. Our list data should be minimalistic, with only necessary information. The suitable fields would be:
1. Name
2. Start locating
3. locations
4. price
5. Duration
6. Metadata(Image etc.)

The rest of the information should be on DetailView.

Comment (awaisdar001) [reply djreact/djangoapps/api/serializers.py]: Let's start with this information and move can optimise in future. :)

**Comment by awaisdar001** [comment]
Perfect thanks :)

**Pull request merged**